### PR TITLE
#1571 Fixed using Response constants in exception codes configuration

### PR DESCRIPTION
**Pull request by edefimov**
According to [configuration](http://symfony.com/doc/master/bundles/FOSRestBundle/4-exception-controller-support.html) we can use constants from Symfony Response class as a value for `exception.codes` configuration section.
However when the bundle is configured in such a way, the constant is not converted into corresponding http status code what leads to error during run time. This PR fixes incorrect behavior and converts constant to value during configuration normalization.

### CODE REVIEW

**Comment by Ener-Getick** [comment]
Could you add tests for that please?

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    private $processor;
+
+    /**
+     * testAcceptsIntegers.
+     */
+    public function testAcceptsIntegers()
+    {
+        $expectedConfig = [
+            md5(microtime()) => mt_rand(200, 599),
```
Comment (Ener-Getick) [inline Tests/DependencyInjection/ConfigurationTest.php]: I think something static would be better :)

File: Tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     * testThatResponseConstantsConvertedToCodes.
+     */
+    public function testThatResponseConstantsConvertedToCodes()
+    {
+        $ref = new \ReflectionClass(Response::class);
+        $expectedResult = [];
+        $config = [];
+
+        foreach ($ref->getConstants() as $constantName => $value) {
+            if (strpos($constantName, 'HTTP_') !== 0) {
```
Comment (Ener-Getick) [inline Tests/DependencyInjection/ConfigurationTest.php]: Same here, no need to test all the constants ;)

File: Tests/DependencyInjection/ConfigurationTest.php (hunk 1 above)
Comment (edefimov) [inline Tests/DependencyInjection/ConfigurationTest.php]: It could be a very relyable test :)

**Comment by Ener-Getick** [comment]
I went through our extension, and in fact it looks like this is already managed [here](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/DependencyInjection/FOSRestExtension.php#L342-L347). Are you sure this doesn't work?

**Comment by edefimov** [comment]
Yes. Well, the whole picture is that on the lines [329-332](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/2.1.0/DependencyInjection/FOSRestExtension.php#L329-L332) the configuration is injected into the service as is and only after that the constants are actually replaced with their values. So, I think the complete fix should remove the substitution logic from extension and move exception class validation from extension into configuration.

**Comment by Ener-Getick** [comment]
No problem for me :) We just need to remove these lines of the extension then.

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -389,15 +390,51 @@ private function addExceptionSection(ArrayNodeDefinition $rootNode)
...
+                                            );
+                                        }
+
+                                        $item = constant('Symfony\Component\HttpFoundation\Response::'.$item);
+                                    }
+
+                                    return $items;
+                                })
                             ->end()
                             ->prototype('scalar')->end()
```
Comment (Ener-Getick) [inline DependencyInjection/Configuration.php]: Can be changed to `int` .

**Comment by Ener-Getick** [comment]
Except my last comment, LGTM :+1:

**Comment by edefimov** [comment]
Done

**Comment by Ener-Getick** [comment]
Thank you @edefimov !

**Pull request merged**